internal/pkg/cli/pipeline_deploy.go

```diff
@@ -60,8 +65,14 @@ type deployPipelineOpts struct {
 	envStore         environmentStore
 	ws               wsPipelineReader
 	codestar         codestar
+	newSvcListCmd    func(io.Writer) actionCommand
+	newJobListCmd    func(io.Writer) actionCommand
 
 	shouldPromptUpdateConnection bool
+
+	// cache variables
+	svcBuffer *bytes.Buffer
+	jobBuffer *bytes.Buffer
```

Comment:
nit: can we make these `io.Writer` as well?

Reply:
Hmm i'm more inclined to keep as is because it is supposed to be `*bytes.Buffer` and we call `Byte()` when we unmarshal the json (we could use Encoder but we'll need to initiate and then call encode).